cnd/src/network/mod.rs

```diff
@@ -136,6 +138,81 @@ impl Swarm {
             local_peer_id,
         })
     }
+
+    /// This is the API for Alice to call in order to execute the appropriate
+    /// communication protocols to announce a swap to Bob (i.e., to send the
+    /// known swap parameters and receive the remaining swap parameters) in
+    /// order to finalize this swap.
+    ///
+    /// This works for all supported combinations on Han, HALight, and Herc20.
+    pub async fn finalize_announce_swap<A, B>(
+        &mut self,
+        _digest: SwapDigest,
+        _announce: AnnounceSwapParameters<A, B>,
+    ) -> anyhow::Result<AnnouncedSwapParameters<A, B>> {
+        // let mut _swarm = self.swarm.lock().await;
+        // swarm.behaviour.finalize_announce_swap(_digest, _announce).await?
+
+        unimplemented!()
+    }
+
+    /// This is the API for Bob to call in order to execute the appropriate
+    /// communication protocols when an announced swap is received from Alice
```

Comment:
I would prefer if the whole communication-cycle happens within the Swarm and does not require two calls.

Reply:
@rishflab and I were just discussing this now.  We could not work it out.
 
Assumption: conversion from `swap_digest` to `swap_id` during the announce protocol requires some logic i.e., to confirm or not to confirm the announced swap requires some logic.  

Question: What does this logic look like?  Does the swarm know everything that is needed by this logic? 

Depending on the answer to the above question, it seems either

1. The swarm **does not** know everything required: Then the inbound event (arrival of the `swap_digest` to Bob's node) has to be propagated up _out_ of the swarm (e.g., by polling the swarm).
2. The swarm **does** know everything and this logic can all be handled within the swarm.  In this case there are also two possibilities, ether the logic is handled in the swarm or in the NetworkBehaviour i.e, either in the swarm (as seen by cnd) or within the `ExpandedSwarm` as seen by the network module.

Reply:
There is no layer above the swarm for incoming events? Or where would you put that logic?

Same as currently, the swarm will need access to the database and other things to spawn the swap later.
We can use the same dependencies to also fetch the necessary data in case of an incoming announce and act accordingly.

Reply:
> There is no layer above the swarm for incoming events? Or where would you put that logic?

I have no better place to suggest, so in the swarm it goes :)

> Same as currently, the swarm will need access to the database and other things to spawn the swap later.
> We can use the same dependencies to also fetch the necessary data in case of an incoming announce and act accordingly.

Cool, I think we are on the same page.